paperqa/contrib/zotero.py

```diff
@@ -109,17 +109,8 @@ def get_pdf(self, item: dict) -> Union[Path, None]:
 
         return pdf_path
 
-    def iterate(
-        self,
-        q: Optional[str] = None,
-        qmode: Optional[str] = None,
-        since: Optional[str] = None,
-        tag: Optional[str] = None,
-        sort: Optional[str] = None,
-        direction: Optional[str] = None,
-        limit: int = 25,
-        start: int = 0,
-    ):
+
+    def iterate(self, limit: int = 25, start: int = 0, collection_name: str = "", **kwargs):
```

Comment:
```suggestion    
    def iterate(
        self,
        limit: int = 25,
        start: int = 0,
        collection_name: Optional[str] = None,
        q: Optional[str] = None,
        qmode: Optional[str] = None,
        since: Optional[str] = None,
        tag: Optional[str] = None,
        sort: Optional[str] = None,
        direction: Optional[str] = None,
    ):
```
I would perhaps do it like this, so autocomplete in IDEs/IPython still works.

Then include a check in the method where if `collection_name` is not None, then all of the query-related arguments must be None (and vice-versa). Does that make sense?

Reply:
Yes. That's good.